app/models/deploy.rb

```diff
@@ -4,7 +4,8 @@ class Deploy < ActiveRecord::Base
   belongs_to :stage, touch: true
   belongs_to :build
   belongs_to :job
-  belongs_to :buddy, class_name: 'User'
+  belongs_to :buddy, -> { unscope(where: "deleted_at") }, class_name: 'User'
+  belongs_to :stage_with_deleted, -> { unscope(where: "deleted_at") }, class_name: 'Stage', foreign_key: 'stage_id'
```

Comment:
It removes the "WHERE stages.deleted_at IS NULL" from the query.  So it would get both non-deleted and deleted stages.  If the desired is deleted stage, it would need unscope(where: "deleted_at").where("stages.deleted_at IS NOT NULL") instead.  

I didn't want to update the regular stages scope, since that would cause some wierd UI bugs, but changing the buddy association was fine, since generally if people are looking back in time, they would still want to know who the buddy is even though the buddy is no longer an active user in Samson.